### PR TITLE
Replace TOC with in-page navigation

### PR DESCRIPTION
**Pull request by nick-mon1**
# Summary

Replace TOC with [in-page navigation](https://designsystem.digital.gov/components/in-page-navigation/).
This is a follow up to #5795.

### Communities Pages

- [Communicators](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/communities/communicators/)
- [Multilingual](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/communities/multilingual/)
- [Plain Language](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/communities/plain-language/)
- [Social Media](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/communities/social-media/)
- [User Experience](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/communities/user-experience/)
- [Web Analytics](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/communities/web-analytics-and-optimization/)
- [Web Managers](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/communities/web-content-managers/)

### About Pages

@ToniBonittoGSA  We have a rule for the in-page navigation that will only show 3 or more navigation links on a page.
The "Write for us" page is the only one that will display sidebar navigation, all others will not.
We can remove this limit if you prefer?

- [About](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/about/)
- [Subscribe](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/about/subscribe/)
- [Write for us](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/about/contribute/)
- [Contact Us](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/about/contact/)

### Community Pages
**Before**
Screen Shot 2023-03-14 at 5 31 38 PM

**After**
Screen Shot 2023-03-14 at 5 31 52 PM

### About Pages

**Before**

Screen Shot 2023-03-15 at 11 17 52 AM
Screen Shot 2023-03-15 at 11 18 34 AM

**After**

Screen Shot 2023-03-15 at 11 17 38 AM
Screen Shot 2023-03-15 at 11 18 25 AM

### CODE REVIEW

File: themes/digital.gov/layouts/communities/single.html [hunk 1]
```diff
@@ -1,154 +1,129 @@
...
-              <div class="community_events">
-                {{- partial "core/get-upcomingevents.html" . -}}
-              </div>
-
-              <div class="Community_conduct">
-                <h2 id="community-conduct">Community Conduct</h2>
-                {{- partial "core/community_pagefooter.html" . -}}
-                <br>
+    <div class="usa-in-page-nav-container">
+      <aside class="usa-in-page-nav" data-title-heading-level="h3" data-root-margin="-350px 0px -350px 0px"></aside>
```
Comment (mejiaj) [inline themes/digital.gov/layouts/communities/single.html:27]: Can we add a comment for what these `data-root-margin` values are doing?

Comment (nick-mon1) [reply themes/digital.gov/layouts/communities/single.html:27]: Good idea, I've added comment explaining why these values are set.

**Review by mejiaj — CHANGES_REQUESTED** [review]
@nick-mon1 noticing a few issues:

### Console error

Certain pages are showing an error in devtools console.

```bash
Uncaught TypeError: Cannot read properties of null (reading 'style')
    at HTMLDocument.<anonymous> 
```

Found on these pages:
- [Guides and resources](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/resources/)
- [Individual guides](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/guides/rpa/)
- [Individual resources](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/topics/21st-century-idea/)

### Inaccurate state in Style guide

The in-page nav state doesn't seem to update on the [styleguide page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-update-in-page-nav/style-guide/).

---

Not a requirement right now, but in the future we should try to create a single template for in-page nav.

**Comment by nick-mon1** [comment]
@mejiaj I've fixed the `console.log` error, but the style flickers on the style-guide page is still there after changing some `IntersectionObserver` settings. It must be related to the content and header structure, will have to document this in detail at some point for some improvements for the `in-page nav` component.

File: themes/digital.gov/src/js/common/in-page-navigation.js [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
-  const navItems = document.querySelectorAll("li.usa-in-page-nav__item:not(.usa-in-page-nav__item--sub-item)");
+  const inPageNavHeader = document.querySelector(".usa-in-page-nav__heading");
+  const navItems = document.querySelectorAll(
+    "li.usa-in-page-nav__item:not(.usa-in-page-nav__item--sub-item)"
+  );
 
-
-
-  if (!relatedItems) {
+  if (inPageNav && navItems.length < 3) {
```
Comment (mejiaj) [inline themes/digital.gov/src/js/common/in-page-navigation.js:9]: @nick-mon1 simplified logic here.

**Review by mejiaj — APPROVED** [review]
Looks good, thanks Nick!

**Review by ToniBonittoGSA — APPROVED** [review]

**Pull request merged**